### PR TITLE
change newline delimiter from "\n" to /\R/

### PR DESCRIPTION
**Pull request by tsuburin**
Fix of https://github.com/markdownlint/markdownlint/issues/237

### CODE REVIEW

File: lib/mdl/doc.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def initialize(text, ignore_front_matter = false)
       else
         @offset = 0
       end
-      @lines = text.split("\n")
+      @lines = text.split(/\R/)
```
Comment (jaymzh) [inline lib/mdl/doc.rb:44]:
```suggestion
      @lines = text.lines
```

**Review by jaymzh — CHANGES_REQUESTED** [review]

**Review by jaymzh — CHANGES_REQUESTED** [review]
Use ruby's platform-agnostic line-splitting please.

**Comment by jamesquilty** [comment]
The change to `text.lines` in line 44 caused a large number of MD009, MD022, MD031, and even an MD035, rule violations when I ran with `mdl --git-recurse --style <my style> .` I checked the reported MD009 violations and they were false positive, so I assume that the remaining reported violations are also false positives introduced by this change.

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
Yeah, @apjanke pointed that out in the associated bug, it's going to not include the newline characters, and if we strip, that'll miss things. So this is probably the right call.

**Comment by jaymzh** [comment]
Can you rebase and force-push, that'll kick travis back into shape:

```
git fetch upstream
git rebase upstream/master
git push --force-with-lease origin doc_lines_regex
```

**Comment by jaymzh** [comment]
Oh nevermind, I managed to kick travis myself

**Comment by jaymzh** [comment]
OK... you need a rebase and then you have some failing tests. :)

**Review by jaymzh — CHANGES_REQUESTED** [review]
Requesting changes until tests pass...

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
Thanks!

**Pull request merged**